crates/ark/src/data_explorer/column_profile.rs

```diff
@@ -0,0 +1,289 @@
+//
+// column_profile.rs
+//
+// Copyright (C) 2024 by Posit Software, PBC
+//
+//
+
+use amalthea::comm::comm_channel::CommMsg;
+use amalthea::comm::data_explorer_comm::ColumnFrequencyTable;
+use amalthea::comm::data_explorer_comm::ColumnHistogram;
+use amalthea::comm::data_explorer_comm::ColumnProfileParams;
+use amalthea::comm::data_explorer_comm::ColumnProfileRequest;
+use amalthea::comm::data_explorer_comm::ColumnProfileResult;
+use amalthea::comm::data_explorer_comm::ColumnProfileSpec;
+use amalthea::comm::data_explorer_comm::ColumnProfileType;
+use amalthea::comm::data_explorer_comm::ColumnSummaryStats;
+use amalthea::comm::data_explorer_comm::DataExplorerFrontendEvent;
+use amalthea::comm::data_explorer_comm::FormatOptions;
+use amalthea::comm::data_explorer_comm::GetColumnProfilesParams;
+use amalthea::comm::data_explorer_comm::ReturnColumnProfilesParams;
+use amalthea::socket::comm::CommSocket;
+use anyhow::anyhow;
+use harp::exec::RFunction;
+use harp::exec::RFunctionExt;
+use harp::tbl_get_column;
+use harp::RObject;
+use harp::TableKind;
+use stdext::unwrap;
+
+use crate::data_explorer::histogram;
+use crate::data_explorer::summary_stats::summary_stats;
+use crate::data_explorer::table::Table;
+use crate::data_explorer::utils::display_type;
+use crate::modules::ARK_ENVS;
+
+pub struct ProcessColumnsProfilesParams {
+    pub table: Table,
+    pub indices: Option<Vec<i32>>,
+    pub kind: TableKind,
+    pub request: GetColumnProfilesParams,
+}
+
+pub async fn handle_columns_profiles_requests(
+    params: ProcessColumnsProfilesParams,
+    comm: CommSocket,
+) -> anyhow::Result<()> {
+    let event = process_columns_profiles_requests(params).await?;
+    let json_event = serde_json::to_value(event)?;
+    comm.outgoing_tx.send(CommMsg::Data(json_event))?;
+    Ok(())
+}
+
+async fn process_columns_profiles_requests(
+    params: ProcessColumnsProfilesParams,
+) -> anyhow::Result<DataExplorerFrontendEvent> {
+    let GetColumnProfilesParams {
+        callback_id,
+        profiles,
+        format_options,
+    } = params.request;
+    let span = tracing::trace_span!("get_profile", ns = callback_id);
+
+    // This is an R thread, so we can actually get the data frame.
+    // If it fails we quickly return an empty result set and end the task.
```

Comment:
Needs a discussion of how this can happen, i.e. when the data explorer instance has been dropped. This is a way of cancelling the task in that case.